src/lib/liquidity-curve.js

```diff
@@ -75,12 +75,22 @@ class LiquidityCurve {
    * @returns {LiquidityCurve}
    */
   combine (curve) {
-    return new LiquidityCurve(
-      this._mapToMax(curve.points)
+    const combined = this._mapToMax(curve.points)
         .concat(curve._mapToMax(this.points))
         .concat(this._crossovers(curve))
         .sort(comparePoints)
-        .filter(omitDuplicates))
+        .filter(omitDuplicates)
+    let prevY = 0
```

Comment:
You assign `prevY` never anything else than 0. I assume you intend to assign the Y value of the previous point and then compare it with next point's Y value.